### PR TITLE
ci: azure: run regression tests (make check) in QEMUv8

### PR DESCRIPTION
**Pull request by jforissier**
Adds a separate job to execute "make check" in the QEMUv8 environment.
The default configuration is tested as well as Normal World
virtualization (XEN_BOOT=y).

The Docker image used for this is jforissier/optee_os_ci:qemuv8_check
on Docker Hub [1]. The Dockerfile is available on Github [2].

Link: [1] https://hub.docker.com/repository/docker/jforissier/optee_os_ci/
Link: [2] https://github.com/jforissier/docker_optee_os_ci/blob/qemuv8_check/Dockerfile
Signed-off-by: Jerome Forissier <jerome@forissier.org>

### CODE REVIEW

**Comment by jforissier** [comment]
Depends on https://github.com/OP-TEE/build/pull/497.

File: .azure-pipelines.yml [hunk 1]
```diff
@@ -174,3 +174,25 @@ jobs:
...
+    pool:
+      vmImage: ubuntu-18.04
+    container:
+      image: jforissier/optee_os_ci:qemuv8_check
+    steps:
+      - script: |
+          set -e -v
+          export LC_ALL=C
+          WD=$(pwd)
+          sudo -E bash -c "cd /root/optee_repo_qemu_v8 && repo sync"
```
Comment (jenswi-linaro) [inline .azure-pipelines.yml:189]: Would it make sense with a `-j` flag to `repo sync`?

Comment (jforissier) [reply .azure-pipelines.yml:189]: This `sync` is not expected to update a lot of things most of the time, but yes it may be worth adding `-j` still.

**Comment by jforissier** [comment]
`xtest` [failure](https://dev.azure.com/OPTEE/optee_os/_build/results?buildId=738&view=logs&j=705748f3-7146-5e86-79af-1c0266d20a8c&t=d3458a7d-1a81-5639-8e63-8927d807e53a&l=7096) with XEN_BOOT=y hopefully fixed by https://github.com/OP-TEE/optee_os/pull/4709.

**Comment by jforissier** [comment]
Update. All tests passed, the new job took slightly less time than the "build all" job so the overall CI duration is unchanged.

**Comment by jenswi-linaro** [comment]
`Acked-by: Jens Wiklander <jens.wiklander@linaro.org>`

**Pull request merged**